### DISCUSSION TITLE
should failed_hosts be included in every MultiResult?

### DISCUSSION THREAD

**Discussion by chatasos** [nornir-automation/nornir#900] [Q&A]
Hi,

Let's assume i have a directory of 1000 hosts and 100 of them are unreachable.

I have noticed that for every router the nornir.core.task.MultiResult "includes" all the failed_hosts. Is there a specific reason for doing so?

```
<class 'nornir.core.task.MultiResult'> ['r10']
│   ├── failed = True
│   ├── failed_hosts = {'r1': MultiResult: [Result: "send_commands_from_file"], 'r2': 
│   │   MultiResult: [Result: "send_commands_from_file"], 'r3': MultiResult: [Result: "send_commands_from_file"], 
│   │   more failed hosts here...
<class 'nornir.core.task.MultiResult'> ['r11']
│   ├── failed = True
│   ├── failed_hosts = {'r1': MultiResult: [Result: "send_commands_from_file"], 'r2': 
│   │   MultiResult: [Result: "send_commands_from_file"], 'r3': MultiResult: [Result: "send_commands_from_file"], 
│   │    more failed hosts here...

```

In the worst case, why have 100 failed hosts being "included" in each one of 1k MultiResults? Does this help in some lookups?
Excuse my ignorance, I'm just trying to understand the reasoning behing the reference of a subset of the inventory on each host result of the inventory.

I'm using https://github.com/packetcoders/nornir-inspect to print the above output, i hope i'm not misinterpreting it.

**Comment by ktbyers** [comment]
@chatasos You probably should either:

1. Direct the question to `nornir-inspect` GitHub
2. Frame the question by using nornir directly and without using `nornir-inspect`
